tests/rptest/services/redpanda.py

```diff
@@ -1931,6 +1964,42 @@ def check_cluster_readiness(self, cluster_name: str,
             f"Cluster {cluster_name} arrived at readyReplicas {expected_replicas}"
         )
 
+    def check_cluster_readiness_operator_v2(self, expected_replicas: int,
+                                            pod_timeout: int):
+        """Checks if the cluster has the expected number of ready replicas."""
+        self.logger.info(
+            f"Waiting for cluster (operator-v2) to have readyReplicas {expected_replicas} with timeout {pod_timeout}"
+        )
+
+        wait_until(
+            lambda: self.cluster_ready_replicas_operator_v2(
+            ) == expected_replicas,
+            timeout_sec=pod_timeout,
+            backoff_sec=1,
+            err_msg=
+            f"Cluster (operator-v2) failed to arrive at readyReplicas {expected_replicas}"
+        )
+
+        self.logger.info(
+            f"Cluster (operator-v2) arrived at readyReplicas {expected_replicas}"
+        )
+
+    def cluster_desired_replicas(self, cluster_name: str):
+        expected_replicas = int(
+            self.kubectl.cmd([
+                'get', 'cluster', cluster_name, '-n=redpanda',
+                "-o=jsonpath='{.status.replicas}'"
+            ]))
+        return expected_replicas
+
+    def cluster_desired_replicas_operator_v2(self):
+        """Return cluster desired replica count for operator v2"""
+        try:
+            rp_statefulset = self.get_redpanda_statefulset()
+            return int(rp_statefulset["status"]["replicas"])
+        except (subprocess.CalledProcessError, KeyError, TypeError):
+            return None
```

Comment:
This is really broad, especially the `CalledProcessError`. Do silently catching all these exceptions and returning have some reasoning.

Also, in the wait until, we compare the return value to an integeger, which will throw if this method ever returns `None`, which is probably not what you wanted (better to throw the original exception in that case!).

Reply:
This try/except block has been removed